src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -63,13 +63,13 @@ private BoundStatement MakeLoweredForm(BoundPatternSwitchStatement node)
                     initialTemp = _factory.SynthesizedLocal(expression.Type, expression.Syntax);
                     result.Add(_factory.Assignment(_factory.Local(initialTemp), expression));
                     expression = _factory.Local(initialTemp);
-                }
 
-                // EnC: We need to insert a hidden sequence point to handle function remapping in case 
-                // the containing method is edited while methods invoked in the expression are being executed.
-                if (!node.WasCompilerGenerated && _localRewriter.Instrument)
-                {
-                    expression = _localRewriter._instrumenter.InstrumentSwitchStatementExpression(node, expression, _factory);
+                    // EnC: We need to insert a hidden sequence point to handle function remapping in case 
+                    // the containing method is edited while methods invoked in the expression are being executed.
+                    if (!node.WasCompilerGenerated && _localRewriter.Instrument)
+                    {
+                        expression = _localRewriter._instrumenter.InstrumentSwitchStatementExpression(node, expression, _factory);
```

Comment:
>expression = _localRewriter._instrumenter.InstrumentSwitchStatementExpression(node, expression, _factory); [](start = 24, length = 106)

Doing instrumentation conditionally based on whether we have a constant value is inconsistent with normal switch. If lowering depends on the ConstantValue, we can either pass it separately, or can move the instrumentation to later point in time, after all analysis is done, right before we put expression to the lowered tree.